applications/CoSimulationApplication/tests/co_simulation_test_factory.py

```diff
@@ -205,6 +206,16 @@ def test_sdof_fsi(self):
 
             self._runTest()
 
+    def test_PFEM_FEM_water_slide_2d(self):
+        if not numpy_available:
+            self.skipTest("Numpy not available")
+        if not have_pfem_fem_dependencies:
+            self.skipTest("PFEM FEM dependencies are not available!")
+
+        with KratosUnittest.WorkFolderScope(".", __file__):
+            self._createTest("pfem_fem_waterslide2d","cosim_pfem_fem_waterslide2d")
+            self._runTest()
+
```

Comment:
There are two leftover files when I run the test locally in my machine, producing the following output:

```
/home/sose2021/Software/Kratos/applications/CoSimulationApplication/tests/co_simulation_test_case.py:33: ResourceWarning: unclosed file <_io.TextIOWrapper name='totalVolumeBeforeMeshing.txt' mode='w' encoding='UTF-8'>
  CoSimulationAnalysis(self.cosim_parameters).Run()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/home/sose2021/Software/Kratos/applications/CoSimulationApplication/tests/co_simulation_test_case.py:33: ResourceWarning: unclosed file <_io.TextIOWrapper name='PfemFluidModelPart.Fluid_number_iterations.dat' mode='w' encoding='UTF-8'>
  CoSimulationAnalysis(self.cosim_parameters).Run()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
ok
```
One of these files is the number of iterations from the print_iteration_number coupling operation. I can remove this from the co_sim_parameters... but I thought the test could take care of this.
The other file is always produced when running PFEM simulations, so there I am also not sure how to proceed.
Am I supposed to manually erase these files with the `tearDownClass()` at the bottom of the test factory?
Also, I put my test into the `TestCoSimulationCases`, because if I tried to put it locally into the nightly, I couldn't make it run.

Finally, there is a lot of output produced when running the test (locally), but only due to the PFEM solver. I don't know why, because I quickly checked the message output in the solver and all of it is also wrapped to an echo level.

Reply:
I fixed it, basically I disabled all the output
Unless necessary for the test we do not write any output

One final concern: There are many warnings (I think from the structural solver), can you please check and fix them
~~~
[WARNING] ResidualBasedBlockBuilderAndSolver: ATTENTION! setting the RHS to zero!
[WARNING] DisplacementCriteria: NaN norm is detected. Setting reference to convergence criteria
~~~

Reply:
@philbucher I addressed this and now shouldn't have these warning messages. I changed the parameter files, not the code itself.